### PR TITLE
Mirror signed release images from GCR to GHCR as part of release with Cloud Build.

### LINKED ISSUE #421: Publish container images to Github container registry
**Description**

There is no official source to grab a container image for Fulcio right now. Having published and signed container images in GHCR will allow for them to be used in the public good instance and also by independent operators instead of building from source.

**Why Github container registry?**

I think github makes the images more easily discoverable for independent operators than google container registry. I'll file a ticket to publish rekor images there to so they match if folks think this is the right approach

**Other considerations**

To support PKCS#11 users we need to have an image with CGO enabled / glibc support (based on e.g distroless debian), but I think its worth also supporting a scratch version

### PR DESCRIPTION
**Pull request by k4leung4**
#### Summary

Copy signed released image from GCR to GHCR using cosign cli copy command.
This will ensure the signature will be the same between the two registry.

@cpanato I had to plumb a new GITHUB_USER env var to allow docker auth to copy image from GCR to GHCR. Not sure if there is a better way or not. Please take a look.

 

#### Ticket Link

Fixes #421 

#### Release Note

```release-note
Signed release images will now be available in GitHub Container Registry
```

### CODE REVIEW

**Review by cpanato — APPROVED** [review]
I will check if the sigstore bot has the token with the write/read the packages enable to be able to push the images to ghcr.io

File: release/README.md [hunk 1]
```diff
@@ -86,6 +86,7 @@ One time setup in ./hack/github-oidc-setup.sh. This is to provide GitHub actions
 	- `_KEY_NAME` key name of your  cosign key.
 	- `_KEY_VERSION` version of the key storaged in KMS. Default `1`.
 	- `_KEY_LOCATION` location in GCP where the key is storaged. Default `global`.
+  - `_GITHUB_USER` GitHub user to authenticate for pushing to GHCR.
```
[the remark this answers is not in the document]
Comment (k4leung4) [reply release/README.md]: fixed. thanks

**Comment by k4leung4** [comment]
> 

thank you for checking.

**Comment by cpanato** [comment]
and the bot have the correct permissions :)

**Pull request merged**